### DISCUSSION TITLE
File resfress

### DISCUSSION THREAD

**Discussion by AntalVincz** [alanbjohnston/CubeSatSim#228] [Q&A]
Hi,

Which process refreshes data in files and what is the cycle?
For example the cw.txt
When the files are refreshed, does the new data come from the payload?

Sorry if this is a stupid question, but I try understanding how to work the system.

73' Tony
M0GLU

**Comment by alanbjohnston** [comment]
Hi Tony,

It isn't a stupid question - it is a little tricky.  

There is C code and Python code running at the same time.  The C code is here https://github.com/alanbjohnston/CubeSatSim/blob/master/main.c and generates the telemetry at regular intervals - this code generates the cw.txt file and touches the cwready file which is a flag.

The python code running is in https://github.com/alanbjohnston/CubeSatSim/blob/master/rpitx.py  It handles the transmitting, and when it sees the cwready file, it transmits the cw.txt info and then deletes the cwready file so that it can detect the next one.

The APRS mode does a similar thing with the t.txt file and the ready file.

In the BPSK and FSK modes, the python code just starts the rpitx processes then blinks the transmit LEDs.

Having these run as separate processes allows for process priority to be set as well as CPU utilization to be easily tracked.  

One day I should write up in the Wiki how the software is architected.

If this doesn't answer your question, let me know and I can explain further.

Alan

**Reply by AntalVincz** [reply]
Hi Alan,

Thank you for your answer.

I understand now.

And if I understand in good, the two separate processes are the cubesatsim and rpitx?

Regards

Tony

Video from my project : https://www.youtube.com/watch?v=LVW-eNgssvE

**Reply by alanbjohnston** [reply]
Yes, that is correct!
